lib/buffer.js

```diff
@@ -65,6 +71,9 @@ function alignPool() {
  * would ever actually be removed.
  **/
 function Buffer(arg, encodingOrOffset, length) {
+  if (!(this instanceof Buffer)) {
+    bufferDeprecationWarning();
+  }
```

Comment:
Instead of this, please use `process.emitWarning()` ...

```js
process.emitWarning(
  'Calling Buffer() without `new` is deprecated. Use Buffer.alloc(), ' +
  'Buffer.allocUnsafe(), Buffer.from(), or new Buffer() instead.',
  'DeprecationWarning'
);
```